src/archivematicaCommon/lib/fileOperations.py

```diff
@@ -394,3 +394,23 @@ def findFileInNormalizationCSV(
                 file=sys.stderr,
             )
             raise FindFileInNormalizatonCSVError(2)
+
+
+def get_extract_dir_name(filename):
+    """
+    Given the name of a compressed file, return the stem directory name into
+    which it should be extracted.
+
+    e.g. transfer1.zip will be extracted into transfer1
+         transfer2.tar.gz will be extracted into transfer2
+    """
+    extract_dir = filename[: filename.rfind(".")]
+
+    # trim off '.tar' if present
+    try:
+        tar_extension_position = extract_dir.rindex(".tar")
+        extract_dir = extract_dir[:tar_extension_position]
+    except ValueError:
+        pass
+
+    return extract_dir
```

Comment:
This code isn't especially robust (I know you didn't write it, but while we're here and adding tests we can improve it). How much that matters depends on where we get these filenames from

Two bugs jump out at me:

* What if the filename is in all caps? e.g. `transfer3.TAR.GZ`. Should be case insensitive.
* What if `.tar` occurs somewhere other than the end of the file? e.g. `my.explanation.of.tar.files.txt`. Should only match it at the end of the filename.

Suggestion:

```python
extract_dir, _ = os.path.splitext(filename)

if extract_dir().endswith(".tar"):
    extract_dir = extract_dir[:-len(".tar")]

return extract_dir
```

Plus some extra test cases.

Reply:
Agreed, now it's extracted as a function it probably is worth improving a little.